### PR TITLE
feat/58-authentication-page-OTP-code-or-verification-code-input-component

### LINKED ISSUE #58: [FEAT] AUTHENTICATION PAGE - OTP / VERIFICATION CODE input component
### Description

_Implement the Inputs that are responsible for accepting OTP code or Verification code according to the design. The inputs should be dynamic and will be responsible for inputing OTP and verification codes._

### Acceptance Criteria

- The inputs have to be done according to the design.
- The inputs should be responsive and not be disrupted on different screens
- Each input should have an onchange function
- The inputs component should return an appended data of all the individual inputs e.g (148372)
- Each input in the component should be able to recieve only one character 
- The next input should automatically be focused when the a character has been typed in the previous input

### Requirements

The following props should be available for the component

1. Normal input props
2. A dynamic number of inputs to display
3. An input data prop to get the inputs data on any input change

### Testing

1. Ensure inputs are displayed properly according to design and requirements
2. Ensure user cannot type in multiple characters in one input
3. Ensure the correct number of inputs are displayed according to what is provided in the props
5. Ensure the appended individual input data is returned on any input change

### Tasks

- [ ] Create the input component
- [ ] Design an input that will be mapped by the number of inputs props. Each input should include the hover and active states
- [ ] Write an onchange function that takes the input data, appends it with other inputs data, stores the appended data in a state and returns the stored data
- [ ] Write unit tests to ensure that the component works as expected
- [ ] Test the component for responsiveness and functionality

### Timeframe

**This task must be completed on or before Friday 11:59 pm as other components depends on this component** 

### Images
Screenshot 2024-07-17 145544
Screenshot 2024-07-17 145603

### Links
Figma Link
[https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=1-18132&m=dev](url)

### PR DESCRIPTION
**Pull request by 64eyes**
# Description

**Closes #58**

# Changes proposed

## What were you told to do?

[FEAT] AUTHENTICATION PAGE - OTP / VERIFICATION CODE input component #58

## What did you do?

Added the OtpInput component that implement the inputs that are responsible for accepting OTP code or verification code according to the design. The inputs are dynamic and responsible for inputing OTP and verification codes.
Also added the OtpInput.test.tsx file that tests out the component.

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [ X] My code follows the code style of this project.
- [ X] This PR does not contain plagiarized content.
- [ X] The title and description of the PR is clear and explains the approach.
- [ X] I am making a pull request against the **dev branch** (left side).
- [ X] My commit messages styles matches our requested structure.
- [ X] My code additions will fail neither code linting checks nor unit test.
- [ X] I am only making changes to files I was requested to.

# Screenshots/Videos

Screenshot (387)

Screenshot of the test
Screenshot (388)

Screenshot of the eslint test
Screenshot (390)

### CODE REVIEW

File: src/test/OtpInput.test.tsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    render(<OtpInput numInputs={6} onChange={vi.fn()} />);
+    const inputs = screen.getAllByRole("textbox");
+
+    for (const input of inputs) {
+      expect(input).toHaveClass("border-[#CBD5E1]");
+      expect(input).toHaveClass("h-[42px]");
+      expect(input).toHaveClass("w-[42px]");
+      expect(input).toHaveClass("md:h-[60px]");
+      expect(input).toHaveClass("md:w-[60px]");
+    }
```
Comment (JulianaSau) [inline src/test/OtpInput.test.tsx]: for loops in tests are not recommended

File: src/test/OtpInput.test.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    render(<OtpInput numInputs={6} onChange={handleChange} />);
+
+    const inputs = screen.getAllByRole("textbox");
+    fireEvent.change(inputs[0], { target: { value: "1" } });
+    fireEvent.change(inputs[1], { target: { value: "2" } });
+
+    expect(handleChange).toHaveBeenLastCalledWith("");
+  });
+
+  it("input fields have the correct styling", () => {
```
Comment (JulianaSau) [inline src/test/OtpInput.test.tsx]: seems like an unnecessary test

**Review by JulianaSau — CHANGES_REQUESTED** [review]

File: src/test/OtpInput.test.tsx (hunk 2 above)
Comment (JulianaSau) [reply src/test/OtpInput.test.tsx]: remove this styling test. its unnecessar

File: src/test/OtpInput.test.tsx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+//     const handleChange = vi.fn();
+//     render(<OtpInput numInputs={6} onChange={handleChange} />);
+
+//     const inputs = screen.getAllByRole("textbox");
+//     fireEvent.change(inputs[0], { target: { value: "1" } });
+//     fireEvent.change(inputs[1], { target: { value: "2" } });
+
+//     expect(handleChange).toHaveBeenLastCalledWith("");
+//   });
+
```
Comment (JulianaSau) [inline src/test/OtpInput.test.tsx]: these seem like tests that make sense. however, if you've commented them out and are not using them delete the code

**Review by JulianaSau — CHANGES_REQUESTED** [review]

**Review by mrcoded — APPROVED** [review]

**Review by JulianaSau — APPROVED** [review]

**Comment by JulianaSau** [comment]
@64eyes ensure your lint checks are passing and that your branch is up to date with the dev branch

**Review by mrcoded — APPROVED** [review]

**Pull request merged**